modules/abstract-cosmos/src/cosmosCoin.ts

```diff
@@ -145,6 +220,91 @@ export class CosmosCoin extends BaseCoin {
     };
   }
 
+  /**
+   * Get the public node url from the Environments constant we have defined
+   */
+  protected getPublicNodeUrl(): string {
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Get account number from public node
+   */
+  protected async getAccountFromNode(senderAddress: string): Promise<request.Response> {
```

Comment:
getAccountFromNode, getBalanceFromNode, getChainIdFromNode, getAccountBalance, getChainId and getAccountDetails are also not needed I guess; We prolly need this only when it comes to WRW recovery

Reply:
Yes i have added it as you will need it as part of recovery task. Adding this in cosmosCoin file will remove duplicacy of code, and the child class can override them in case they have different implementations.